### PR TITLE
Added Prettier as Script and pre-commit hook

### PR DESCRIPTION
**Pull request by AlokTakshak**

### CODE REVIEW

**Comment by AlokTakshak** [comment]
@s-yadav  @ameerthehacker

**Comment by s-yadav** [comment]
The tab spacing doesn't look accurate. It seems to be 4 instead of 2. Can you check that.

**Comment by AlokTakshak** [comment]
> The tab spacing doesn't look accurate. It seems to be 4 instead of 2. Can you check that.
 
to me it looks 2 space only in vs code.

**Comment by s-yadav** [comment]
I think it used tab instead of spaces. Can you check if we can use space instead.

**Comment by AlokTakshak** [comment]
> I think it used tab instead of spaces. Can you check if we can use space instead.

useTabs ->  Indent lines with tabs instead of spaces.
tabWidth -> Specify the number of spaces per indentation-level.

so are using usetabs -> true and tabWidth=2

**Comment by s-yadav** [comment]
Usetabs should be false in that case.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "useTabs": false,
+    "trailingComma": "all",
+    "semi": true,
+    "singleQuote": true,
+    "jsxBracketSameLine": true,
+    "jsxSingleQuote": true
+    }
```
Comment (itaditya) [inline .prettierrc]: format this file and add a newline in the end

File: example/App.js [hunk 2]
```diff
@@ -8,37 +8,37 @@ import CreatePortalExample from './createPortalExample';
 import SVGExample from './SVGExample';
 import LazySuspenseExample from './lazySuspenseExample';
 
-export default function App () {
+export default function App() {
   return (
     <div>
-      <div className="wrapper">
+      <div className='wrapper'>
```
Comment (itaditya) [inline example/App.js]: Most commonly the prop values are kept `double quotes` and the js part is kept single quotes like this

```js
const a = 'hello';
return (
  <div className="wrapper" style={{ backgroundColor: 'black' }}>
    {a}
  </div>
);
```

@s-yadav what do you think?

**Pull request closed without merging**